uktrade/great-cms
# GP2-451: Case study component

This PR contains backend and frontent for the Case Study feature mentioned in the ticket.

The backend work:

* Adds a new CaseStudy snippet, following the spec in the ticket, which can be added to a DetailPage just like the ContentModule used to be able to
* Adds two custom tag types, specifically for personalisation - HS-code tags and Country code tags (both just strings)
* Uses these new custom tag types on the CaseStudy snippet, so it's possible to select a case study via these tags
* Removes the ContentModule from the DetailPage's Streamfield, so it won't appear any more
* Adds temporary/placeholder/good-enough-to-test-drive logic to support filtering for a Case Study by HS code (via `?hs-tag=XXXXX`) or country code (via `?country-tag=YYYYY`) or both (via `?hs-tag=XXX&country-tag=YYYY`)
* Roughs in FE content for @lewis-coulson-dit to work with

The frontend work:

* Builds out the new case study module using components from storybook
* (more detail to come from @lewis-coulson-dit if needed)


## To test drive

* Add a new Snippet, picking the CaseStudy option
  * Confirm that to save it, you need to have exactly the following (please play around): 
    * company name
    * summary paragraph
    * one media block in the body, with at least one image or video, and max two. 
    * content in the rich-text area
* Tags are not needed to save the Snippet, but confirm you can add them to both tag fields. Add a couple, save the page and re-edit. Confirm that tags created for one field do NOT autocomplete for the other field.
* Save two snippets - one with a video and an image. Give BOTH Snippets a HS code of `HS1234`, give one of these snippets a country code of DE and save it and another Snippet with a country code of FR and save it. Remember which you saved last.
* Add the CaseStudy component to a DetailPage/lesson page and save it.

**To confirm the temporary tag logic is working:** 
* Browse to that detail page. Add ?hs-tag=HS1234 to the URL - you should see the case study which was saved most recently, including its media content
* Change the querystring to ?country-tag=FR and you should see the case study associated with that tag; now try DE instead of FR
* Note that providing both querystrings at once will not work right now, but that is OK because this logic will be replaced by a proper algorithm very soon.

## Checklist

 - [X] Ticket exists in Jira  https://uktrade.atlassian.net/browse/GP2-451 
 - [ ] [Changelog](CHANGELOG.md) entry added. --> TO COME AFTER FRONTEND

 - [ ] (if is a UI change) Includes screenshot(s) - ideally before and after, but at least after
 - [X] How to test locally, including how to set up appropriate data
 
 
## Screenshots (Backend)

![Screenshot 2020-10-01 at 15 59 54](https://user-images.githubusercontent.com/101457/94860994-bd8be200-042e-11eb-9c44-0b05e77238a5.png)
![Screenshot 2020-10-01 at 15 59 57](https://user-images.githubusercontent.com/101457/94860997-bebd0f00-042e-11eb-842f-e7e27de766db.png)
![Screenshot 2020-10-01 at 16 03 47](https://user-images.githubusercontent.com/101457/94861003-bf55a580-042e-11eb-98d0-eaabbf09c316.png)
![Screenshot 2020-10-01 at 16 09 09](https://user-images.githubusercontent.com/101457/94861049-d4323900-042e-11eb-8e91-b9cc192fd98d.png)
![Screenshot 2020-10-01 at 16 09 28](https://user-images.githubusercontent.com/101457/94861052-d5636600-042e-11eb-8f95-99835c524cf2.png)
![Screenshot 2020-10-01 at 16 45 18](https://user-images.githubusercontent.com/101457/94861054-d5636600-042e-11eb-8b83-12ef2b0259ec.png)

![Screenshot 2020-10-07 at 09 32 27](https://user-images.githubusercontent.com/101457/95309012-7fede580-0882-11eb-9493-e081371ab14f.png)
![Screenshot 2020-10-07 at 09 33 02](https://user-images.githubusercontent.com/101457/95309016-811f1280-0882-11eb-89df-32fe7d567a7e.png)
![Screenshot 2020-10-07 at 09 46 49](https://user-images.githubusercontent.com/101457/95309019-81b7a900-0882-11eb-8e91-3e90de33e531.png)
![Screenshot 2020-10-07 at 09 46 54](https://user-images.githubusercontent.com/101457/95309021-82503f80-0882-11eb-83b1-805bcd282620.png)
